File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/adapter/GuardPlayer.java
package jv.com.tencent.cain.design_pattern.adapter;

/**
 * @author cainjiang
 * @date 2020/4/15
 */
public class GuardPlayer extends Player {

    public GuardPlayer(String name) {
        super(name);
    }

    @Override
    public void attack() {
        System.out.println("GuardPlayer# " + name + " attack ---");
    }

    @Override
    public void defense() {
        System.out.println("GuardPlayer# " + name + " defense ---");
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/adapter/GuardPlayer.kt
package kt.com.tencent.cain.desigin_pattern.adapter

/**
 * @author cainjiang
 * @date 2020/4/15
 */
class GuardPlayer(name: String) : Player(name) {

    override fun attack() {
        println("GuardPlayer# $name attack ---")
    }

    override fun defense() {
        println("GuardPlayer# $name defense ---")
    }

}